target/linux/ipq40xx/mikrotik/base-files/etc/hotplug.d/firmware/11-ath10k-caldata

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+[ -e /lib/firmware/$FIRMWARE ] && exit 0
+
+. /lib/functions/caldata.sh
+
+board=$(board_name)
+
+case "$FIRMWARE" in
+"ath10k/pre-cal-ahb-a000000.wifi.bin")
+	case "$board" in
+	mikrotik,routerboard-wap-g-5hacd2hnd)
+		wlan_data="/sys/firmware/mikrotik/hard_config/wlan_data"
+		( [ -f "$wlan_data" ] && caldata_sysfsload_from_file "$wlan_data" 0x0 0x2f20 ) || \
+		( [ -d "$wlan_data" ] && caldata_sysfsload_from_file "$wlan_data/data_0" 0x0 0x2f20 )
```

Comment:
Consider creating a function for that, if you expected additional similar entries:

```
caldata_mikrotik_ath10k() {
	local offset=$1
	local size=$2
	local offset2=$3
	local filename=$4

	local wlan_data="/sys/firmware/mikrotik/hard_config/wlan_data"
	( [ -f "$wlan_data" ] && caldata_sysfsload_from_file "$wlan_data" $offset $size ) || \
	( [ -d "$wlan_data" ] && caldata_sysfsload_from_file "$wlan_data/$filename" $offset2 $size )
}

caldata_mikrotik_ath10k 0x0 0x2f20 0x0 data_0
```

Alternatively, one could just move wlan_data out of the case and keep the two remaining lines without a function for now.